meson.build

```diff
@@ -311,7 +311,7 @@ if get_option('build_backends')
       deps += [ opencl_framework ]
       has_opencl = true
 
-  elif opencl_lib.found()
+  elif opencl_lib.found() and cc.has_header('CL/opencl.h', args: '-I' + get_option('opencl_include')[0])
```

Comment:
Maybe make it also resilient to future additions of include directories, in the spirit of https://github.com/LeelaChessZero/lc0/blob/06748a4a35b297164e36400a576bc64031fa0415/meson.build#L222
Just an idea, I imagine someone editing `meson_options.txt` later won't recall or even know that this check only considers the first include directory.

Reply:
We use the same trick in <https://github.com/LeelaChessZero/lc0/blob/master/meson.build#L160>. 
Maybe we should change the option type from `array` to `string` when only a single argument is supported.

Reply:
I did make `opencl_include` a string option with the latest commit. There are a few more candidates but I want to consider them separately